src/Timer/Timers.php

```diff
@@ -77,32 +66,34 @@ public function isEmpty()
 
     public function tick()
     {
-        $time = $this->updateTime();
-        $timers = $this->timers;
-        $scheduler = $this->scheduler;
-
-        while (!$scheduler->isEmpty()) {
-            $timer = $scheduler->top();
+        // ensure timers are sorted so we can execute in order
+        if (!$this->sorted) {
```

Comment:
You have the exact same logic at line 54. You could create a private method `ensureScheduleIsSorted()` or something similar in order to do this checking and perform the sorting operation.

Reply:
Fair point! This is the very core of the event loop, so this is the critical code path and I would rather avoid any additional function calls. What do you think about this? :+1: